### PR TITLE
11957 hardware breakpoint set

### PR DESCRIPTION
**Pull request by AsuNa-jp**
## Change Summary
**This PR adds a new field which indicates whether a hardware breakpoint is set for a thread**. 

Protections team is interested in this information because, for example, hardware breakpoints are used for patch-less hooking to bypass security products/features such as AMSI. 

* Example POC
  * https://gist.github.com/CCob/fe3b63d80890fafeca982f76c8a3efdf

### Sample values
see `package/endpoint/data_stream/alerts/sample_event.json`

## Release Target
8.8.0

### For mapping changes:

- [x] I ran `make` after making the schema changes, and committed all changes
- [x] If these field(s) are "exception"-able, I made a companion PR to Kibana adding it (see [Readme](https://github.com/elastic/endpoint-package#exceptionable))
- [x] If this is a `metadata` change, I also updated both transform destination schemas to match

### For Transform changes:

- [ ] The new transform successfully starts in Kibana
- [ ] The corresponding transform destination schema was updated if necessary

### CODE REVIEW

File: custom_schemas/custom_process.yml [hunk 1]
```diff
@@ -188,6 +188,14 @@
       description: The sha256 of the call_stack_final_user_module.
       example: "d25ff1e6c6460a7f9de39198d182058c1712726008d187e1953b83abe977e4a0"
 
+    - name: thread.Ext.hardware_breakpoint_set
+      level: custom
+      type: boolean
+      description:  >
+        Check whether a hardware breakpoint is set for the thread. 
+        Include this field, only if the hardware breakpoint is setted. 
+      example: "true"
```
Comment (jdu2600) [inline custom_schemas/custom_process.yml:197]: This generates a non-fatal warning - "Short descriptions must be single line, and under 120 characters"
You can explicitly set a short description with `short:`

```suggestion
      short: Whether a hardware breakpoint was set for the thread. 
      description:  >
        Whether a hardware breakpoint was set for the thread. 

        This field is omitted if false.
      example: "true" 
```

Comment (AsuNa-jp) [reply custom_schemas/custom_process.yml]: Thanks! Fixed as suggested!

**Review by gabriellandau — APPROVED** [review]
LGTM

**Comment by gabriellandau** [comment]
Green build

**Comment by AsuNa-jp** [comment]
> LGTM
Thank you very much of fixing this issue, Gabe!

**Comment by AsuNa-jp** [comment]
Hi @pzl @gergoabraham ! Could you take a look when you get a chance?

**Review by pzl — APPROVED** [review]
Looks good!

Sorry for the wait, but thanks for the bump, this is good to merge

**Comment by AsuNa-jp** [comment]
@pzl Thanks for reviewing!

**Pull request merged**